### PR TITLE
tuxedo-drivers: 3.2.14 -> 4.5.0

### PR DESCRIPTION
**Pull request by MinerSebas**
## Description of changes

This updates the tuxedo-keyboard package to 4.5.0, and renames the package to its new upstream name.
While updating the nixos module, I also removed parts of the option description that were already removed in the 3.2.0 release.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MinerSebas** [comment]
I tested it on my Tuxedo Pulse Gen1 with the 6.6 Linux Kernel, but i would appreciate feedback for newer Tuxedo Models. 
From the changelog that would be at least Pulse Gen4, Sirius Gen2, Pulse Gen3 and Stellaris Gen6.

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  meta = {
+    broken = stdenv.isAarch64 || (lib.versionOlder kernel.version "5.5");
+    description = "Keyboard and hardware I/O driver for TUXEDO Computers laptops";
+    homepage = "https://gitlab.com/tuxedocomputers/development/packages/tuxedo-drivers";
+    license = lib.licenses.gpl3Plus;
+    longDescription = ''
+      This driver provides support for Fn keys, brightness/color/mode for most TUXEDO
+      keyboards (except white backlight-only models).
+
+      Can be used with the "hardware.tuxedo-keyboard" NixOS module.
```
Comment (mrcjkb) [inline pkgs/os-specific/linux/tuxedo-drivers/default.nix:50]:
```suggestion
      Can be used with the "hardware.tuxedo-drivers" NixOS module.
```

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    for module in clevo_acpi.ko clevo_wmi.ko \
+                  ite_829x/ite_829x.ko ite_8291/ite_8291.ko ite_8291_lb/ite_8291_lb.ko ite_8297/ite_8297.ko \
+                  tuxedo_compatibility_check/tuxedo_compatibility_check.ko tuxedo_io/tuxedo_io.ko tuxedo_keyboard.ko \
+                  tuxedo_nb02_nvidia_power_ctrl/tuxedo_nb02_nvidia_power_ctrl.ko \
+                  tuxedo_nb04/tuxedo_nb04_kbd_backlight.ko tuxedo_nb04/tuxedo_nb04_keyboard.ko tuxedo_nb04/tuxedo_nb04_power_profiles.ko \
+                  tuxedo_nb04/tuxedo_nb04_sensors.ko tuxedo_nb04/tuxedo_nb04_wmi_ab.ko tuxedo_nb04/tuxedo_nb04_wmi_bs.ko \
+                  tuxedo_nb05/tuxedo_nb05_ec.ko tuxedo_nb05/tuxedo_nb05_fan_control.ko tuxedo_nb05/tuxedo_nb05_kbd_backlight.ko \
+                  tuxedo_nb05/tuxedo_nb05_keyboard.ko tuxedo_nb05/tuxedo_nb05_power_profiles.ko tuxedo_nb05/tuxedo_nb05_sensors.ko \
+                  uniwill_wmi.ko; do
+        mv src/$module $out/lib/modules/${kernel.modDirVersion}
```
Comment (mrcjkb) [inline pkgs/os-specific/linux/tuxedo-drivers/default.nix:35]:
```suggestion
        install -Dm444 src/$module -t $out/lib/modules/${kernel.modDirVersion}
```

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    broken = stdenv.isAarch64 || (lib.versionOlder kernel.version "5.5");
+    description = "Keyboard and hardware I/O driver for TUXEDO Computers laptops";
+    homepage = "https://gitlab.com/tuxedocomputers/development/packages/tuxedo-drivers";
+    license = lib.licenses.gpl3Plus;
+    longDescription = ''
+      This driver provides support for Fn keys, brightness/color/mode for most TUXEDO
```
Comment (mrcjkb) [inline pkgs/os-specific/linux/tuxedo-drivers/default.nix:47]: question: Is this description still accurate?

The [features list](https://github.com/tuxedocomputers/tuxedo-drivers?tab=readme-ov-file#features) seems to suggest no.

**Review by mrcjkb — CHANGES_REQUESTED** [review]
Looks mostly good to me. 
Will test this when I am near my tuxedo device.

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix (hunk 3 above)
Comment (MinerSebas) [reply pkgs/os-specific/linux/tuxedo-drivers/default.nix:47]: Only the `Hardware I/O driver for TUXEDO Control Center` is missing, but the mentioned TUXEDO Control Center isn't even packaged here, and is used by tuxedo_rs instead.

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                  uniwill_wmi.ko; do
+        install -Dm444 src/$module -t $out/lib/modules/${kernel.modDirVersion}
+    done
+
+    runHook postInstall
+  '';
+
+  meta = {
+    broken = stdenv.isAarch64 || (lib.versionOlder kernel.version "5.5");
+    description = "Keyboard and hardware I/O driver for TUXEDO Computers laptops";
```
Comment (mrcjkb) [inline pkgs/os-specific/linux/tuxedo-drivers/default.nix:43]:
```suggestion
    description = "Drivers for several platform devices for TUXEDO notebooks meant for the DKMS";
```

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix (hunk 3 above)
Comment (mrcjkb) [reply pkgs/os-specific/linux/tuxedo-drivers/default.nix:47]: Then I would suggest to add "Hardware IO driver" without "for Tuxedo Control Center"

**Review by mrcjkb — CHANGES_REQUESTED** [review]

**Comment by LemmusLemmus** [comment]
Nice work!

There is also this open pull request that intends to replace tuxedo-keyboard with tuxedo-drivers: #293017. I guess you can compare your implementation with theirs?

Either way, once this has been merged, would it make sense to enable the package by default in [nixos-hardware](https://github.com/NixOS/nixos-hardware/tree/master/tuxedo)?

**Comment by xaverdh** [comment]
I can confirm that this works on my Aura Gen1, using `tuxedo-rs`

File: pkgs/os-specific/linux/tuxedo-drivers/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  ];
+
+  makeFlags = [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}"
+
+    for module in clevo_acpi.ko clevo_wmi.ko \
```
Comment (xaverdh) [inline pkgs/os-specific/linux/tuxedo-drivers/default.nix:26]: https://github.com/NixOS/nixpkgs/pull/293017 uses INSTALL_MOD_PATH here, which I find nice

**Comment by panicgh** [comment]
I cherry-picked this onto nixos-24.05 and successfully used it for the past ~4 weeks with tuxedo-rs on a Pulse 14 Gen 4.

**Comment by ClockGen** [comment]
Can this be merged? Currently can't use tuxedo-keyboard in linux > 6.9 due to this https://github.com/NixOS/nixpkgs/issues/329383 (the patch from issue should probably be added too).

**Comment by MinerSebas** [comment]
As was too distracted to actively work on this, i am closing this PR.
For those that still wants this merged, look at either #336633 (my preference) or #293017 that do the same thing.

**Pull request closed without merging**